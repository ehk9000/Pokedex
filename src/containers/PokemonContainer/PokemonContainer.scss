main {
  display: flex;
  flex-flow: row wrap;
  justify-content: space-evenly;
}

.loading {
  margin-top: 200px;
  height: 300px;
}