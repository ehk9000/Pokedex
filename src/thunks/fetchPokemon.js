import { getPokemon, setLoading, setError} from '../actions/index';

export const fetchPokemon = () => {
  return async (dispatch) => {
    const url = 'https://pokeapi.co/api/v2/generation/1/';
    try {
      dispatch(setLoading(true));

      const response = await fetch(url);

      if (!response.ok) {
        throw Error(response.statusText);
      }

      const pokemon = await response.json();
      dispatch(setLoading(false));
    } catch (error) {
      dispatch(setError(error.message));
    }
  }
}

