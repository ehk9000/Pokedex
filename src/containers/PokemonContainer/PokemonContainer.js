
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { fetchPokemon } from '../../thunks/fetchPokemon';

export class PokemonContainer extends Component {
   componentDidMount() {
     this.props.fetchPokemon();
   }
   
  render() {
    return (
      <main>
  
      </main>
    )
  }
}

export const mapStateToProps = ({ pokemon }) => ({
  pokemon
})

export const mapDispatchToProps = dispatch => ({
  fetchPokemon: () => dispatch(fetchPokemon())
});

export default connect(mapStateToProps, mapDispatchToProps)(PokemonContainer);