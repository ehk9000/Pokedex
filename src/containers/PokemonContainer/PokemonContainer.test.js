import React from 'react';
import { shallow } from 'enzyme';
import { PokemonContainer, mapStateToProps, mapDispatchToProps } from './PokemonContainer';
import { fetchPokemon } from '../../thunks/fetchPokemon';

jest.mock("../../thunks/fetchPokemon");

describe('PokemonContainer', () => {
  let wrapper;
  let mockLocation;
  beforeEach(() => {
    mockLocation = { pathname: '/'};
    wrapper = shallow( <PokemonContainer 
      fetchPokemon={fetchPokemon}
      location={mockLocation}/> )
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});