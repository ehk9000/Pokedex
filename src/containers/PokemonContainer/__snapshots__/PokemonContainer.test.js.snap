// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonContainer should render correctly 1`] = `ShallowWrapper {}`;
